### ISSUE TITLE
Not compatible with jest version 28

### ISSUE THREAD

**Issue by MarkLyck** [omrilotan/isbot#183]
Steps to reproduce
-------
`import isbot from 'isbot'` anywhere in a file that's tested.
install and use `jest@28` to test the file.

Expected behaviour
-------
The isbot import should work.

Actual behaviour
-------
Jest throws an error about unexpected token inside isbot.

```
Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /Users/marklyck/colony/colony-frontend/node_modules/.pnpm/isbot@3.4.6/node_modules/isbot/index.mjs:328
    export { isbot as default };
    ^^^^^^

    SyntaxError: Unexpected token 'export'

      1 | import LogRocket from 'logrocket'
    > 2 | import isbot from 'isbot'
        | ^
```

Additional details
-------
Jest@28 started supporting exports in package.json;
https://jestjs.io/docs/upgrading-to-jest28#packagejson-exports

This is why this breaks, and it suggests submitting an issue to `isbot` to fix the exports in this package to be correct.

**Comment by omrilotan** [comment]
I'm afraid the issue may with your build system. Would you like to share some project details?
Node version, Jest configuration, and any additional build configuration (webpack etc).

I've successfully tested using Jest with no build, using both native ESM and commonjs.
Please review Pull Request #184 


![](https://user-images.githubusercontent.com/516342/167447492-846fbca1-bbc5-4ce6-85ed-34279a6c460d.png)

**Comment by omrilotan** [comment]
## Update

Changing 
```diff
- "browser": "./index.mjs"
+ "browser": {
+   "import": "./index.mjs",
+   "require": "./index.js"
+ }
```
Did not help. Jest resolve the module without modules in VM flag.

I'm open to suggestions

**Comment by wadehammes** [comment]
This is failing for me using NextJS 12.1.6 with vanilla configuration (using their built in SWC compiler)

![image](https://user-images.githubusercontent.com/222170/167877868-a543ed38-0e6a-4640-ad45-d6ee5c86958b.png)

**Comment by wadehammes** [comment]
This helped me: https://stackoverflow.com/questions/60714101/how-to-setup-jest-with-node-modules-that-use-es6

Adding `transformIgnorePatterns: ["<rootDir>/node_modules/(?!isbot)"],` to my custom Jest config allowed tests to pass with Jest 28:

![image](https://user-images.githubusercontent.com/222170/167881047-a1224a71-a20c-4be6-8806-6dd6b7986065.png)

**Comment by wadehammes** [comment]
I spoke to soon, apparently my yarn had updated on this branch to a previous version and that's why it passed. my fix above does not work with Jest 28. My apologies. Still experiencing the export issue.

**Comment by omrilotan** [comment]
I've added babel transpiling, removed VM module flag, and typescript tests.
I'll need to see your transpiling setup. Are you using webpack or something on top of everything?
Have a look at #186

**Comment by wadehammes** [comment]
Im using a standard NextJS setup. There is some webpack config but mainly for Sentry.

Next 12.1.5 using their SWC compiler (no babel)

jest.config.ts
```
// jest.config.ts
import type { Config } from "@jest/types";
import nextJest from "next/jest";

// Sync object
const customJestConfig: Config.InitialOptions = {
  verbose: true,
  testPathIgnorePatterns: ["<rootDir>/.next/", "<rootDir>/node_modules/"],
  setupFiles: ["<rootDir>/.jest/setEnvVars.ts"],
  setupFilesAfterEnv: ["<rootDir>/.jest/setupTests.ts"],
  moduleDirectories: ["node_modules", "<rootDir>"],
  testEnvironment: "jest-environment-jsdom",
  transformIgnorePatterns: ["<rootDir>/node_modules/(?!isbot)"],
};

// Providing the path to your Next.js app which will enable loading next.config.js and .env files
const createJestConfig = nextJest({ dir: "./" })(customJestConfig);

module.exports = async () => {
  // Create Next.js jest configuration presets
  const jestConfig = await createJestConfig();

  // Custom `moduleNameMapper` configuration
  const moduleNameMapper = {
    ...jestConfig.moduleNameMapper,
    "\\.(css|less|scss|sass)$": "identity-obj-proxy",
  };

  return { ...jestConfig, moduleNameMapper };
};
```

tsconfig
```
{
  "compilerOptions": {
    "allowJs": false,
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "declaration": false,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "resolveJsonModule": true,
    "rootDir": ".",
    "skipLibCheck": true,
    "strict": true,
    "target": "es2015",
    "typeRoots": [
      "node_modules/@types"
    ],
    "incremental": true,
    "paths": {
      "@mui/styled-engine": ["./node_modules/@mui/styled-engine-sc"]
    }
  },
  "include": [
    "cssprops.d.ts",
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    ".jest/setupTests.ts",
    "i18next.config.ts"
  ],
  "exclude": [
    "node_modules",
    "next.config.js"
  ]
}
```

**Comment by omrilotan** [comment]
Still can't reproduce. Would you like to try and test using?
```json
{
  "test": "NODE_OPTIONS=--experimental-vm-modules jest"
}
```

**Comment by wadehammes** [comment]
That works fine locally but in Github actions:

![image](https://user-images.githubusercontent.com/222170/167964277-1395325d-affc-4544-be33-d68189ea2fe8.png)

Any ideas?

**Comment by omrilotan** [comment]
@wadehammes Good news,
Once I've defined jest environment as jsdom - I was able to replicate your issue.

If you could please test the next version using "next" tag - we can see that it solved the issue for you and release a stable version:

```
npm i isbot@next
```

**Comment by wadehammes** [comment]
@omrilotan this resolved the issue, thank you!

**Comment by omrilotan** [comment]
> @omrilotan this resolved the issue, thank you!

Brilliant.Ill release a version in about 30 minutes. Thank you for taking the time

**Comment by omrilotan** [comment]
Resolved in version 3.5.0
